addons/test_mail/tests/test_mail_channel.py

```diff
@@ -394,3 +394,18 @@ def test_user_moderation_counter(self):
         })
         self.assertEqual(self.user_employee.moderation_counter, 2)
         self.assertEqual(self.user_employee_2.moderation_counter, 0)
+
+
+@tagged("mention")
+class TestChannelMention(TestMailCommon):
+
+    def test_mention_suggestions(self):
+        mention = self.env.user.partner_id.get_mention_suggestions('')
+        self.assertEqual(len(mention), 2)
+        self.assertEqual(len(mention[0]), 4)
+        self.assertEqual(len(mention[1]), 8)
+
+    def test_mention_odoobot(self):
+        mention = self.env.user.partner_id.get_mention_suggestions('odoobot')
+        self.assertEqual(mention[1][0]['id'], 2)
+        self.assertEqual(mention[1][0]['name'], 'OdooBot')
```

Comment:
This should not be part of `test_mail_channel` file, as it's just testing a method on partner.

Considering the currently existing test files, I think a new one for this would be appropriate.

eg. `test_partner_mention`


Also these tests need adaptation, as it is it not not clear what is their goal, and where the result is coming from.

I think I retract my idea of adding these test on this PR. You can just remove them here and keep the diff for later.

We should make a new task for that purpose to properly and fully testing mentions python side instead.